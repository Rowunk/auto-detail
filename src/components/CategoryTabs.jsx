// src/components/CategoryTabs.jsx
import React from 'react';

const categories = [
  { key: 'all',      label: '📋 Vše'        },  // ← new tab
  { key: 'wash',     label: '🚿 Mytí'       },
  { key: 'exterior', label: '🚗 Exteriér'   },
  { key: 'wheels',   label: '⚙️ Kola'       },
  { key: 'interior', label: '🪑 Interiér'   },
  { key: 'protection', label: '🛡️ Ochrana' },
  { key: 'restoration', label: '🔧 Opravy'  },
  { key: 'specialty',  label: '⭐ Speciální'}
];

/**
 * Horizontal scrollable pill-tabs.
 * Props: active (string), onChange(key)
 */
export default function CategoryTabs({ active, onChange }) {
  return (
    <div className="flex space-x-2 overflow-x-auto px-2 py-1">
      {categories.map(cat => (
        <button
          key={cat.key}
          onClick={() => onChange(cat.key)}
          className={`whitespace-nowrap px-3 py-1 rounded-full text-sm font-semibold transition
            ${active === cat.key
              ? 'bg-blue-500 text-white shadow'
              : 'bg-white text-gray-700 border border-gray-300 dark:bg-primary dark:text-gray-200 dark:border-gray-600'}`}
        >
          {cat.label}
        </button>
      ))}
    </div>
  );
}
