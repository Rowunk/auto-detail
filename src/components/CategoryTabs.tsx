// src/components/CategoryTabs.tsx
import React from 'react';
import PropTypes from 'prop-types';
import { CategoryTabsProps } from '../types/props';
import { ServiceCategory } from '../types';

type CategoryInfo = {
  key: ServiceCategory;
  label: string;
};

const categories: CategoryInfo[] = [
  { key: 'all',      label: '📋 Vše'        },
  { key: 'wash',     label: '🚿 Mytí'       },
  { key: 'exterior', label: '🚗 Exteriér'   },
  { key: 'wheels',   label: '⚙️ Kola'       },
  { key: 'interior', label: '🪑 Interiér'   },
  { key: 'protection', label: '🛡️ Ochrana' },
  { key: 'restoration', label: '🔧 Opravy'  },
  { key: 'specialty',  label: '⭐ Speciální'}
];

/**
 * Horizontal scrollable tabs for filtering services by category.
 *
 * @param {CategoryTabsProps} props - Component props
 * @param {ServiceCategory} props.active - Currently selected category
 * @param {Function} props.onChange - Callback when category is changed
 * @returns {React.ReactElement} Category tabs component
 * 
 * @example
 * <CategoryTabs 
 *   active="wash" 
 *   onChange={(category) => setActiveCategory(category)} 
 * />
 */
export default function CategoryTabs({ active, onChange }: CategoryTabsProps): React.ReactElement {
  return (
    <div className="flex space-x-2 overflow-x-auto px-2 py-1">
      {categories.map(cat => (
        <button
          key={cat.key}
          onClick={() => onChange(cat.key)}
          className={`whitespace-nowrap px-3 py-1 rounded-full text-sm font-semibold transition
            ${active === cat.key
              ? 'bg-blue-500 text-white shadow'
              : 'bg-white text-gray-700 border border-gray-300 dark:bg-primary dark:text-gray-200 dark:border-gray-600'}`}
        >
          {cat.label}
        </button>
      ))}
    </div>
  );
}

CategoryTabs.propTypes = {
  active: PropTypes.oneOf([
    'all', 'wash', 'exterior', 'wheels', 'interior', 
    'protection', 'restoration', 'specialty'
  ]).isRequired,
  onChange: PropTypes.func.isRequired
};