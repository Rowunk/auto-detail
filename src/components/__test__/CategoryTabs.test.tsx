// src/components/__test__/CategoryTabs.test.tsx

import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import CategoryTabs from '../CategoryTabs';
import type { ServiceCategory } from '../../types';

describe('CategoryTabs', () => {
  const mockOnChange = jest.fn();

  afterEach(() => {
    mockOnChange.mockReset();
  });

  const allCategories: ServiceCategory[] = [
    'favorites', 'all', 'wash', 'exterior', 'wheels',
    'interior', 'protection', 'restoration', 'specialty'
  ];

  it('renders all category buttons with correct labels', () => {
    render(<CategoryTabs active="all" onChange={mockOnChange} />);
    expect(screen.getByText('⭐ Oblíbené')).toBeInTheDocument();
    expect(screen.getByText('📋 Vše')).toBeInTheDocument();
    expect(screen.getByText('🚿 Mytí')).toBeInTheDocument();
    expect(screen.getByText('🚗 Exteriér')).toBeInTheDocument();
    expect(screen.getByText('⚙️ Kola')).toBeInTheDocument();
    expect(screen.getByText('🪑 Interiér')).toBeInTheDocument();
    expect(screen.getByText('🛡️ Ochrana')).toBeInTheDocument();
    expect(screen.getByText('🔧 Opravy')).toBeInTheDocument();
    expect(screen.getByText('⭐ Speciální')).toBeInTheDocument();
  });

  it('highlights the active category tab', () => {
    render(<CategoryTabs active="wash" onChange={mockOnChange} />);
    const labels = [
      '⭐ Oblíbené', '📋 Vše', '🚿 Mytí', '🚗 Exteriér',
      '⚙️ Kola', '🪑 Interiér', '🛡️ Ochrana', '🔧 Opravy', '⭐ Speciální'
    ];
    labels.forEach(label => {
      const btn = screen.getByText(label);
      if (label === '🚿 Mytí') {
        expect(btn).toHaveClass('bg-blue-500', { exact: false });
      } else {
        expect(btn).not.toHaveClass('bg-blue-500', { exact: false });
      }
    });
  });

  it('calls onChange with correct category key when a tab is clicked', () => {
    render(<CategoryTabs active="all" onChange={mockOnChange} />);
    const wheelsBtn = screen.getByText('⚙️ Kola');
    fireEvent.click(wheelsBtn);
    expect(mockOnChange).toHaveBeenCalledWith('wheels');
  });

  it('calls onChange for all category tabs', () => {
    render(<CategoryTabs active="all" onChange={mockOnChange} />);
    const labelMap: Record<ServiceCategory, string> = {
      favorites: '⭐ Oblíbené',
      all: '📋 Vše',
      wash: '🚿 Mytí',
      exterior: '🚗 Exteriér',
      wheels: '⚙️ Kola',
      interior: '🪑 Interiér',
      protection: '🛡️ Ochrana',
      restoration: '🔧 Opravy',
      specialty: '⭐ Speciální'
    };
    allCategories.forEach(cat => {
      const btn = screen.getByText(labelMap[cat]);
      fireEvent.click(btn);
      expect(mockOnChange).toHaveBeenCalledWith(cat);
    });
  });

  it('matches snapshot (visual regression)', () => {
    const { container } = render(<CategoryTabs active="all" onChange={mockOnChange} />);
    expect(container).toMatchSnapshot();
  });
});
