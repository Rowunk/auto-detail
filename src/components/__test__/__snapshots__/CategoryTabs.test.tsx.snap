// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CategoryTabs matches snapshot (visual regression) 1`] = `
<div>
  <div
    class="flex space-x-2 overflow-x-auto px-2 py-1"
  >
    <button
      class="whitespace-nowrap px-3 py-1 rounded-full text-sm font-semibold transition
            bg-white text-gray-700 border border-gray-300 dark:bg-primary dark:text-gray-200 dark:border-gray-600"
    >
      ⭐ Oblíbené
    </button>
    <button
      class="whitespace-nowrap px-3 py-1 rounded-full text-sm font-semibold transition
            bg-blue-500 text-white shadow"
    >
      📋 Vše
    </button>
    <button
      class="whitespace-nowrap px-3 py-1 rounded-full text-sm font-semibold transition
            bg-white text-gray-700 border border-gray-300 dark:bg-primary dark:text-gray-200 dark:border-gray-600"
    >
      🚿 Mytí
    </button>
    <button
      class="whitespace-nowrap px-3 py-1 rounded-full text-sm font-semibold transition
            bg-white text-gray-700 border border-gray-300 dark:bg-primary dark:text-gray-200 dark:border-gray-600"
    >
      🚗 Exteriér
    </button>
    <button
      class="whitespace-nowrap px-3 py-1 rounded-full text-sm font-semibold transition
            bg-white text-gray-700 border border-gray-300 dark:bg-primary dark:text-gray-200 dark:border-gray-600"
    >
      ⚙️ Kola
    </button>
    <button
      class="whitespace-nowrap px-3 py-1 rounded-full text-sm font-semibold transition
            bg-white text-gray-700 border border-gray-300 dark:bg-primary dark:text-gray-200 dark:border-gray-600"
    >
      🪑 Interiér
    </button>
    <button
      class="whitespace-nowrap px-3 py-1 rounded-full text-sm font-semibold transition
            bg-white text-gray-700 border border-gray-300 dark:bg-primary dark:text-gray-200 dark:border-gray-600"
    >
      🛡️ Ochrana
    </button>
    <button
      class="whitespace-nowrap px-3 py-1 rounded-full text-sm font-semibold transition
            bg-white text-gray-700 border border-gray-300 dark:bg-primary dark:text-gray-200 dark:border-gray-600"
    >
      🔧 Opravy
    </button>
    <button
      class="whitespace-nowrap px-3 py-1 rounded-full text-sm font-semibold transition
            bg-white text-gray-700 border border-gray-300 dark:bg-primary dark:text-gray-200 dark:border-gray-600"
    >
      ⭐ Speciální
    </button>
  </div>
</div>
`;
